src/sentry/static/sentry/app/stores/projectsStore.tsx

```diff
@@ -1,10 +1,45 @@
 import Reflux from 'reflux';
-import each from 'lodash/each';
 
+import {Project, Team} from 'app/types';
 import ProjectActions from 'app/actions/projectActions';
 import TeamActions from 'app/actions/teamActions';
 
-const ProjectsStore = Reflux.createStore({
+type State = {
+  projects: Project[];
+  loading: boolean;
+};
+
+type StatsData = Record<string, Project['stats']>;
+
+/**
+ * Attributes that need typing but aren't part of the external interface,
+ */
+type Internals = {
+  itemsById: Record<string, Project>;
+  loading: boolean;
+};
+
+type ProjectsStoreInterface = {
+  init: () => void;
+  reset: () => void;
+  loadInitialData: (projects: Project[]) => void;
+  onStatsLoadSuccess: (data: StatsData) => void;
+  onChangeSlug: (prevSlug: string, newSlug: string) => void;
+  onCreateSuccess: (project: Project) => void;
+  onUpdateSuccess: (data: Partial<Project>) => void;
+  onDeleteTeam: (slug: string) => void;
+  onRemoveTeam: (teamSlug: string, projectSlug: string) => void;
+  onAddTeam: (team: Team, projectSlug: string) => void;
+  removeTeamFromProject: (teamSlug: string, project: Project) => void;
+  getWithTeam: (teamSlug: string) => Project[];
+  getAll: () => Project[];
+  getById: (id: string | undefined) => Project | undefined;
+  getBySlug: (slug: string | undefined) => Project | undefined;
+  getBySlugs: (slug: string[]) => Project[];
+  getState: (slugs?: string[]) => State;
```

Comment:
```suggestion
  init: () => void;
  reset: () => void;
  loadInitialData: (projects: Project[]) => void;
  onStatsLoadSuccess: (data: StatsData) => void;
  onChangeSlug: (prevSlug: string, newSlug: string) => void;
  onCreateSuccess: (project: Project) => void;
  onUpdateSuccess: (data: Partial<Project>) => void;
  onDeleteTeam: (slug: string) => void;
  onRemoveTeam: (teamSlug: string, projectSlug: string) => void;
  onAddTeam: (team: Team, projectSlug: string) => void;
  removeTeamFromProject: (teamSlug: string, project: Project) => void;
  getWithTeam: (teamSlug: string) => Project[];
  getAll: () => Project[];
  getBySlugs: (slug: string[]) => Project[];
  getState: (slugs?: string[]) => State;
  getById?: (id?: string) => Project;
  getBySlug?: (slug?: string) => Project;
```

Reply:
just noticed that the function shouldn't be of type undefined, but it's return type